### DISCUSSION TITLE
How can I have cypress post a complete report

### DISCUSSION THREAD

**Discussion by bezba** [cypress-io/cypress#9627] [Questions and Help]
Hi Currently I am posting to Slack when the test fails. but i wanna be able to add the report at the end of the test as well. So when all the feature files are done and we reported all the broken test. I wanna be able to report this as well 

>        Spec                                              Tests  Passing  Failing  Pending  Skipped  
  

> ┌─────────────────────────────────────────────────────────
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✔  176.feature                  00:31        1        1        -        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✔  185.feature                  00:07        1        1        -        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✔  186.feature                  00:11        1        1        -        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✖  187.feature                  06:26        7        6        1        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✖ 194.feature                  03:29        1        -        1        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✔ 199.feature                  00:34        1        1        -        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✔  200.feature                  00:34        1        1        -        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✔  201.feature                  00:30        1        1        -        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✔  203.feature                  01:00        1        1        -        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✔  205.feature                  01:04        1        1        -        -        - │
>   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
>   │ ✖  227.feature                  01:47        1        -        1        -        - │
>   ──────────────────────────────────────┘
>     ✖  3 of 29 failed (10%)                     34:22       40      37        3        -        -  
> 
> 

This what I have right now. and every time test its finishes it posts everything for success . I am not sure how I can access that above report

````
after(() => {
  let reportedTests;
  let url = Cypress.env('SLACK_WEBHOOK_URL'); 
  let time = new Date().toLocaleString();
  const findTests = (suite) => {
    suite.tests.forEach((test) => {
      if (test.state === "failed" && test._retries >= 3) {
        reportedTests = {

          "attachments": [
            {
              "color": "#FF0000",
              "fields": [
                {
                  "title": "The following test has failed ",
                  "value": test.title,
                  "short": true,
                },
              ],
            },
          ],
        };
        cy.request({
          method: "POST",
          url: url,
          followRedirect: false,
          headers: {
            "Content-Type": "application/json",
          },
          body: reportedTests
        });
      }
    });
    if (suite.suites.length < 1) {
//I need this to only happen at the end when all the tests are done and give me the above report
  reportedTests = {
    "text": "======  :checkered_flag: :checkered_flag: ==== " + time,
  };
  cy.request({
    method: "POST",
    url: url,
    followRedirect: false,
    headers: {
      "Content-Type": "application/json",
    },
    body: reportedTests
  });
      return;
    }
    suite.suites.forEach((nestedSuite) => {
      findTests(nestedSuite);
    });
  };
  cy.state("runnable").parent.suites.forEach((suite) => {
    findTests(suite);
  });
 
});
```
`

**Comment by bahmutov** [comment]
If you need the full test report with all test results, use our NPM module API https://on.cypress.io/module-api to run Cypress. Then send the results object anywhere you want. See some examples of using module API in https://glebbahmutov.com/blog/cy-metaprogramming/

**Reply by bezba** [reply]
@bahmutov  how can I POST after all the tests are done? A single post. Is there anyway that I can chat with you?
